src/components/RecipesList.js

```diff
@@ -25,6 +27,7 @@ class RecipesList extends React.Component {
   };
 
   onHandleSubmit(event) {
+    this.setState({showDataLoading: true});
```

Comment:
May want to separate this out into its own function that gets triggered. Seems like this handle submit method is handling quite a lot of logic now.

Reply:
This did cross my mind but I reasoned that it was one line in the `onHandleSubmit` that would just be replaced by another one line that would call a function. The benefit of another separate function would be that you could give a reviewer a clear idea about what was being done - but again I felt this one line is pretty clear.
Later on in the method I opted to have one call that `setState` as thought could be more efficient. These two factors combined were what led me with the other current approach.

Reply:
I agree with your argument and I think it's fine. I would argue that the point of breaking it out into a separate function is less about line count and more about expandability. If it was in a separate function then you could easily change something if you needed to. Totally fine the way it is for now.